service/controller/chart/v1/resource/status/create.go

```diff
@@ -70,6 +73,12 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		if key.IsCordoned(cr) {
 			status = releaseStatusCordoned
 			reason = key.CordonReason(cr)
+		} else if cc.Status.Reason != "" {
+			// A Helm release exists for this chart CR but data was also added
+			// to the controller context. We do this if something goes wrong
+			// outside of Helm such as pulling the chart tarball. So we ensure
+			// both messages are included in the CR status.
+			reason = fmt.Sprintf("Helm reason: %s Operator reason: %s", releaseHistory.Description, cc.Status.Reason)
```

Comment:
I added more comments here so there is context. I also extended the reason field to show both ours and the helm reason.

Helm does a pretty good job of making these human readable. So I think this helps.